tcf_website/views/profile.py

```diff
@@ -51,4 +53,21 @@ def profile(request):
 @login_required
 def reviews(request):
     """User reviews view."""
-    return render(request, 'reviews/user_reviews.html')
+    # Handled separately because it requires joining 1 more table (i.e. Vote)
+    upvote_stat = Review.objects.filter(user=request.user).aggregate(
+        total_review_upvotes=Count('vote', filter=Q(vote__value=1)),
+    )
+    # Get other statistics
+    other_stats = User.objects.filter(id=request.user.id).aggregate(
+        total_reviews_written=Count('review'),
+        average_review_rating=(
+            Avg('review__instructor_rating') +
+            Avg('review__enjoyability') +
+            Avg('review__recommendability')
+        ) / 3,
```

Comment:
Not super important but I wonder if this logic would be better as a function in the User model

Reply:
There was [a similar discussion](https://docs.djangoproject.com/en/3.1/topics/db/managers/) before, but the reason why our code was very slow in general is that we used functions (for individual instance) and iterated over querysets to invoke those functions. A faster and more DB-friendly approach is to use `JOIN`s and aggregation, which translates to annotations in Django ORM. I've been looking into [managers and querysets](https://docs.djangoproject.com/en/3.1/topics/db/managers/) to pull out repeated annotations, but it has low priority because it's just for refactoring.

Reply:
If this is the only remaining issue, it seems like we can squash-merge this PR?

Reply:
Gotcha, makes sense! Will merge for you